basics.md

```diff
@@ -0,0 +1,345 @@
+Instead, that is a property of the _site_ to which the mutation refers, in this case,
+site ID 0 (which happens to be at position 257):
+
+```{code-cell} ipython3
+print(ts.site(0))  # For convenience, the Python API returns the mutations at this site too
+```
+
+In the plot above, since the the only mutation is above node 8 and has a
+{attr}`~Mutation.derived_state` of "G", we know that the samples descending from node
+8 in the first tree (sample genomes 2, 4, and 5) have a "G" at {attr}`~Site.position` 257,
+while the others have the {attr}`~Site.ancestral_state` of "T". This means that Ada is
+homozygous for "T", Bob is heterozygous "T|G", and Cat is homozygous for "G".
+In other words the ancestral state and the details of any mutations at that site,
+when coupled with the tree topology at the site {attr}`~Site.position` is sufficient to
+define the allelic state possessed by each sample.
+
+Note that even though the genome is 1000 base pairs long, the tree sequence only contains
+a single site, because we usually only bother defining *variable* sites in a tree
+sequence (e.g. positions seen in studies to have samples possessing different alleles at
+that genomic location). It is perfectly possible to have a site with no mutations
+(a "monomorphic" site) but such sites are not normally very useful.
+
+### Provenance
+
+:::{todo}
+What is provenance
+:::
+
+
+## Concepts
+
+There are some basic population genetic concepts which can be helpful with thinking about
+tree sequences. For reference, here is the tree sequence topology that we have been using:
+
+```{code-cell} ipython3
+:"tags": ["hide-input"]
+from IPython.display import SVG
+SVG(ts.draw_svg(y_axis=True, y_gridlines=True, time_scale="rank", style=".mut {display: none}"))
+```
+
+
+### Tree changes, ancestral recombinations, and SPRs
+
+Note that adjacent trees in this tree sequence differ by a single "tree edit" or SPR
+(subtree-prune-and-regraft) operation: the subtree defined by node 8 in the first
+tree has been pruned and regrafted onto the branch from 10 to 3, to create the
+second tree.
+
+The second and third trees have the same topology, but differ because their
+ultimate coalesence happened in a different ancestor (this is easy to see in a simulation,
+but hard to detect in real data). The SPR that created the third tree must involve pruning
+either the subtree below node 7 or the subtree below node 9 and grafted it higher up on
+the same lineage. In this case it turns out to be impossible to tell just from the tree
+sequence which of those two SPRs actually happened in the simulation.
+
+In general, each detectable recombination occurring in ancestral history results in a
+single SPR in the tree sequence. If recombination breakpoints occurs at unique
+positions (an "infinite sites" model of breakpoints), then the number of trees in a tree
+sequence equals the number of ancestral recombination events minus one. If recombinations
```

Comment:
I'm either missing something here or this should be that the number of trees is equal to the number of recombinations plus one? :confused:

Reply:
Well spotted. Copied from the old workshop material, where it's presumably wrong too!